warehouse/query-core/src/main/java/datawave/query/iterator/logic/TermFrequencyIndexIterator.java

```diff
@@ -184,15 +184,24 @@ public void next() throws IOException {
             }
             
             DatawaveKey key = new DatawaveKey(top);
+            Key nextTop = null;
             
             for (int i = 0; i < 256 && source.hasTop() && key.getFieldName().compareTo(field) < 0; ++i) {
                 source.next();
-                key = new DatawaveKey(source.getTopKey());
+                
+                nextTop = source.getTopKey();
+                if (nextTop == null)
+                    break;
```

Comment:
I wouldn't hold it up over this, and some could argue it's too esoteric, but it seems like this could actually be a good use of a [labelled continue](https://www.developer.com/java/data/understanding-the-java-labeled-statement.html) statement. Label the outer while loop, change this to `continue <label>` and remove the addition at line 202.